File: Swift/power-of-three.swift
class Solution {
    func isPowerOfThree1(_ n: Int) -> Bool {
        var x = 1

        while x != n {
            if x < n {
                x *= 3
            } else {
                return false
            }
        }

        return true
    }

    func isPowerOfThree2(_ n: Int) -> Bool {
        return n > 0 && 1162261467 % n == 0
    }
}

File: Python/power-of-three.py
import math

class Solution:
    def isPowerOfThree1(self, n):
        """
        :type n: int
        :rtype: bool
        """
        x = 1

        while x != n:
            if x < n:
                x *= 3
            else:
                return False

        return True

    def isPowerOfThree2(self, n):
        """
        :type n: int
        :rtype: bool
        """
        maxPow3 = 3 ** int(math.log(0x7fffffff) / math.log(3))
        return n > 0 and maxPow3 % n == 0
